uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Wrapper/LeaveRequestFieldsVisibility.php

```diff
@@ -0,0 +1,135 @@
+      'request_requires_contact_id' => TRUE
+    ],
+    'getbreakdown' => [
+      'handler' => 'CRM_HRLeaveAndAbsences_API_Handler_GetBreakDownFieldPermissions'
+    ],
+  ];
+
+  /**
+   * @param $apiRequest
+   * @return mixed
+   */
+  public function fromApiInput($apiRequest) {
+    if ($this->canHandleApiInput($apiRequest)) {
+      $returnParams = $this->getRequestReturnParameter($apiRequest);
+
+      if(!empty($returnParams)) {
+        $apiRequest['params']['return'] = $returnParams;
+      }
+    }
+
+    return $apiRequest;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function toApiOutput($apiRequest, $result) {
+    if ($this->canHandleAPiOutput($apiRequest)) {
+      $handler = $this->getHandler($apiRequest);
+      $handler->process($result);
+    }
+
+    return $result;
+  }
+
+  /**
+   * Returns the handler for the API action.
+   *
+   * @param $apiRequest
+   *
+   * @return mixed
+   */
+  private function getHandler($apiRequest) {
+    $handler = $this->apiActions[$apiRequest['action']]['handler'];
+    $leaveRequestRights = new LeaveRequestRightsService(new LeaveManagerService());
+
+    return new $handler($apiRequest, $leaveRequestRights);
+  }
+
+  /**
+   * Checks whether the request can be handled or not.
+   * 
+   * @param $apiRequest
+   *
+   * @return bool
+   */
+  private function canHandleTheRequest($apiRequest) {
+    $isTargetAction = array_key_exists($apiRequest['action'], $this->apiActions);
+    return $apiRequest['entity'] === 'LeaveRequest' && $isTargetAction;
+  }
+
+  /**
+   * Checks If the API request output can be handled.
+   *
+   * @param $apiRequest
+   *
+   * @return bool
+   */
+  private function canHandleAPiOutput($apiRequest) {
+    return $this->canHandleTheRequest($apiRequest);
+  }
+
+  /**
+   * Checks If the API request input can be handled.
+   *
+   * @param array $apiRequest
+   *
+   * @return bool
+   */
+  private function canHandleApiInput($apiRequest) {
+    $canHandleRequest = $this->canHandleTheRequest($apiRequest);
+    $requiresContactID = !empty($this->apiActions[$apiRequest['action']]['request_requires_contact_id']);
+
+    return $canHandleRequest && $requiresContactID;
+  }
+
+  /**
+   * Ensures that the return parameter for the API input has all the dependencies
+   * needed to make the API handlers work. In this case the contact_id should be
+   * present in the list of fields to be returned in the result set so that the API
+   * handlers that requires this can return accurate results.
+   *
+   * @param array $apiRequest
+   *
+   * @return array
+   */
+  private function getRequestReturnParameter($apiRequest) {
+    $options = _civicrm_api3_get_options_from_params($apiRequest['params']);
+    $returnParams = array_keys($options['return']);
+
+    if (!empty($returnParams)) {
+      if (FALSE === array_search('contact_id', $returnParams)) {
+        $returnParams[] = 'contact_id';
+      }
+    }
+
+    return $returnParams;
```

Comment:
We need to be a bit smarter here. If a request didn't include the contact_id in the return fields, it means they don't want that and will be quite confused when they see the contact_id among the returned fields. It's fine to add the fields to the return params, but when processing the api output we need to remove it in case it not specified